### DISCUSSION TITLE
Executing model produces raw output

### DISCUSSION THREAD

**Discussion by bmharper** [fcakyon/yolov5-pip#166] [General]
When I run this code:

```python
model = yolov5.load("yolov5m6.pt")
y = model(x)
```
Where x is a tensor with shape (8, 3, 1280, 1280)
Then y is a tensor with shape (8, 102000, 10)
It looks to me like I'm getting the raw output of the model, but the documentation says that one should be getting the post-processed output here.

**Comment by fcakyon** [comment]
Your model output sizes seem to be not normal. Which version are you using?

https://github.com/fcakyon/yolov5-pip#use-from-python

**Comment by bmharper** [comment]
I'm using the latest YOLOv5 master branch from https://github.com/ultralytics/yolov5  (commit 32a92185738c93e5f0b0f6971de0812cd6fd5f34)

Do I need to train using this pip package?

**Reply by bmharper** [reply]
In the readme, it shows that the output has a ".pred" member, but my output is just a plain PyTorch tensor.

**Reply by fcakyon** [reply]
what is your yolov5-pip version?

**Reply by fcakyon** [reply]
when model is not autoshape, then output is plain tensor.

**Reply by fcakyon** [reply]
try using latest yolov5-pip.

**Reply by bmharper** [reply]
My yolov5 pip version is 6.2.3, so I am using the latest.
My code looks exactly like the code in the readme, i.e.:
```python
import yolov5

# load pretrained model
model = yolov5.load('yolov5s.pt')
#....
# perform inference
results = model(img)
```
But my results do not match what the readme describes.

**Reply by fcakyon** [reply]
i am trying but not able to recreate the issue, can you provide me some sample weight file so that I can recreate the issue?

**Reply by fcakyon** [reply]
continuous integration tests are also passing: https://github.com/fcakyon/yolov5-pip/blob/3abe7cf9e5880e1c21d2b24bbab2cfc6244082c6/tests/test_yolov5.py#L148

**Reply by fcakyon** [reply]
have you also tried 6.2.2? Is the same error present with older versions?

**Reply by bmharper** [reply]
My model is too big... I'll train a smaller one and upload it here. I'll also try 6.2.2 and let you know....

**Comment by bmharper** [comment]
Hi,
I've found the cause - it is these lines in `common.py`:

![image](https://user-images.githubusercontent.com/876013/198540547-8f5e38f1-4f4f-4a0a-9c7e-cdd42128ace6.png)

The key issue is that when you feed `AutoShape` a `tensor`, then it does not run the post-processing NMS etc. I'll query the canonical YOLOv5 repo for why this decision exists, or if they'd be open to a flag that controls this behaviour.

**Reply by fcakyon** [reply]
ok thanks for pointing that out 👍

**Comment by bmharper** [comment]
I have opened a PR https://github.com/ultralytics/yolov5/pull/10139 which allows one to perform post-processing on Tensor inputs. I'll comment here again if that PR gets merged.

**Reply by fcakyon** [reply]
thanks for the update 👍
